crates/wasmparser/src/binary_reader.rs

```diff
@@ -1001,6 +1001,23 @@ impl<'a> BinaryReader<'a> {
     {
         let code = self.read_var_u32()?;
         Ok(match code {
+            0x14 => {
+                let ht = HeapType::from_reader(self)?;
+                visitor.visit_ref_test_non_null(ht)
+            }
+            0x15 => {
+                let ht = HeapType::from_reader(self)?;
+                visitor.visit_ref_test_nullable(ht)
+            }
+            0x16 => {
+                let ht = HeapType::from_reader(self)?;
+                visitor.visit_ref_cast_non_null(ht)
+            }
+            0x17 => {
+                let ht = HeapType::from_reader(self)?;
+                visitor.visit_ref_cast_nullable(ht)
+            }
```

Comment:
I think these should be possible to do `visitor.visit_ref_cast_nullable(self.read()?)` to be a bit more compact